photon-server/src/main/java/org/photonvision/common/hardware/HardwareManager.java

```diff
@@ -61,27 +61,46 @@ private HardwareManager(HardwareConfig hardwareConfig) {
         CustomGPIO.setConfig(hardwareConfig);
         MetricsBase.setConfig(hardwareConfig);
 
-        statusLED = new StatusLED(hardwareConfig.statusRGBPins);
+        statusLED =
+                hardwareConfig.statusRGBPins.size() == 3
+                        ? new StatusLED(hardwareConfig.statusRGBPins)
+                        : null;
         visionLED =
-                new VisionLED(
-                        hardwareConfig.ledPins,
-                        hardwareConfig.ledPWMFrequency,
-                        (hardwareConfig.ledPWMRange != null && hardwareConfig.ledPWMRange.size() == 2)
-                                ? hardwareConfig.ledPWMRange.get(1)
-                                : 0);
+                hardwareConfig.ledPins.isEmpty()
+                        ? null
+                        : new VisionLED(
+                                hardwareConfig.ledPins,
+                                hardwareConfig.ledPWMFrequency,
+                                (hardwareConfig.ledPWMRange != null && hardwareConfig.ledPWMRange.size() == 2)
+                                        ? hardwareConfig.ledPWMRange.get(1)
+                                        : 0);
 
         ledModeEntry = NetworkTablesManager.getInstance().kRootTable.getEntry("ledMode");
         ledModeEntry.setNumber(VisionLEDMode.VLM_DEFAULT.value);
-        ledModeListener = new NTDataChangeListener(ledModeEntry, visionLED::onLedModeChange);
+        ledModeListener =
+                visionLED == null
+                        ? null
+                        : new NTDataChangeListener(ledModeEntry, visionLED::onLedModeChange);
 
         Runtime.getRuntime().addShutdownHook(new Thread(this::onJvmExit));
 
+        if (visionLED != null)
+            visionLED.setBrightness(
+                    ConfigManager.getInstance().getConfig().getHardwareSettings().ledBrightnessPercentage);
+
         // Start hardware metrics thread (Disabled until implemented)
         // if (Platform.isLinux()) MetricsPublisher.getInstance().startTask();
     }
 
+    public void setBrightnessPercent(int percent) {
+        ConfigManager.getInstance().getConfig().getHardwareSettings().ledBrightnessPercentage = percent;
+        if (visionLED != null) visionLED.setBrightness(percent);
+        ConfigManager.getInstance().requestSave();
+        logger.info("Setting led brightness to " + percent + "%");
+    }
+
     private void onJvmExit() {
```

Comment:
We should also call the [shutdown ](https://github.com/PhotonVision/photonvision/blob/master/photon-server/src/main/java/org/photonvision/common/hardware/GPIO/GPIOBase.java#L74) methods on the LEDs

Reply:
What exactly does the shutdown method do?